src/ctx.cpp

```diff
@@ -2072,7 +2091,19 @@ llvm::Value *FunctionEmitContext::LoadInst(llvm::Value *ptr, const char *name) {
     }
 
     AddDebugPos(inst);
-    return inst;
+
+    llvm::Value *loadVal = inst;
+    // bool type is stored as i8. So, it requires some processing.
+    if ((type != NULL) && (type->IsBoolType())) {
+        if (CastType<AtomicType>(type) != NULL)
+            loadVal = SwitchBoolSize(loadVal, inst->getType(), type->LLVMType(g->ctx));
+        if ((CastType<VectorType>(type) != NULL) && (type->IsBoolType())) {
```

Comment:
I think "else if" here is more appropriate.
Also, please use {} even for single line if, it's less error prone.